edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/AuthorizationConfiguration.cs

```diff
@@ -0,0 +1,144 @@
+    ///
+    /// This object is being eventually constructed from the EdgeHub twin's desired properties.
+    /// See <see cref="EdgeHubDesiredProperties"/> for DTO.
+    /// </summary>
+    public class AuthorizationConfiguration : List<Statement>, IEquatable<AuthorizationConfiguration>
+    {
+        public bool Equals(AuthorizationConfiguration other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return Enumerable.SequenceEqual(this, other);
+        }
+
+        public override bool Equals(object obj)
+            => this.Equals(obj as AuthorizationConfiguration);
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                int hash = 17;
+                hash = this.Aggregate(hash, (acc, item) => (acc * 31 + item.GetHashCode()));
+                return hash;
+            }
+        }
+    }
+
+    public class Statement : IEquatable<Statement>
+    {
+        [JsonConstructor]
+        public Statement(IList<string> identities, IList<Rule> allow, IList<Rule> deny)
+        {
+            this.Identities = identities;
+            this.Allow = allow;
+            this.Deny = deny;
+        }
+
+        [JsonProperty(PropertyName = "identities")]
+        public IList<string> Identities { get; }
+
+        [JsonProperty(PropertyName = "allow")]
+        public IList<Rule> Allow { get; }
+
+        [JsonProperty(PropertyName = "deny")]
+        public IList<Rule> Deny { get; }
+
+        public bool Equals(Statement other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return Enumerable.SequenceEqual(this.Identities, other.Identities)
+                && Enumerable.SequenceEqual(this.Allow, other.Allow)
+                && Enumerable.SequenceEqual(this.Deny, other.Deny);
+        }
+
+        public override bool Equals(object obj)
+            => this.Equals(obj as Statement);
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                int hash = 17;
+                hash = this.Identities.Aggregate(hash, (acc, item) => (acc * 31 + item.GetHashCode()));
+                hash = this.Allow.Aggregate(hash, (acc, item) => (acc * 31 + item.GetHashCode()));
+                hash = this.Deny.Aggregate(hash, (acc, item) => (acc * 31 + item.GetHashCode()));
+                return hash;
+            }
+        }
+    }
+
+    public class Rule : IEquatable<Rule>
+    {
+        [JsonConstructor]
+        public Rule(IList<string> operations, IList<string> resources)
```

Comment:
nit: IEnumerable can be used instead of IList, as we don't care about the count or the order.

Reply:
Good point, but I'll need count later when I implement validation.